### PR TITLE
Implement header forwarding disallow list feature for ext_proc filter.

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
This is a follow up commit for : https://github.com/envoyproxy/envoy/pull/28468.

It is to implement header forwarding disallow list feature for ext_proc filter.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by yanavlasov — DISMISSED** [review]

File: source/extensions/filters/http/ext_proc/ext_proc.h [hunk 1]
```diff
@@ -145,10 +149,10 @@ class FilterConfig {
     const std::string final_prefix = absl::StrCat(prefix, "ext_proc.", filter_stats_prefix);
     return {ALL_EXT_PROC_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
   }
-  const std::vector<Matchers::StringMatcherPtr> initHeaderMatchers(
-      const envoy::extensions::filters::http::ext_proc::v3::ExternalProcessor& config) {
+  const std::vector<Matchers::StringMatcherPtr>
+  initHeaderMatchers(const envoy::type::matcher::v3::ListStringMatcher header_list) {
```
Comment (tyxia) [inline source/extensions/filters/http/ext_proc/ext_proc.h:153]: nit:  const &

**Review by tyxia — APPROVED** [review]
LGTM, nice work

File: source/extensions/filters/http/ext_proc/ext_proc.h (hunk 1 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/ext_proc.h]: good catch! done

**Comment by yanjunxiang-google** [comment]
@envoyproxy/api-shepherds PTAL

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**